src/client/theme-default/composables/outline.ts

```diff
@@ -197,6 +197,7 @@ export function useActiveAnchor(
 
     if (activeLink) {
       activeLink.classList.add('active')
+      activeLink.scrollIntoView({ behavior: 'smooth', block: 'center' })
```

Comment:
PR preview
![image](https://github.com/vuejs/vitepress/assets/59066707/c7ebcd20-e60a-423e-86ef-d15a4778c6a3)

Fixed
![image](https://github.com/vuejs/vitepress/assets/59066707/bb61cc02-19fb-4f20-83ad-3f9d8f0c16f2)

we also need to fix some edge cases like scroll to 0 when there is no active link, I'd also scroll only when it is out of viewport